Library/Homebrew/download_strategy.rb

```diff
@@ -424,27 +424,12 @@ def resolve_url_basename_time_file_size(url)
       url = url.sub(%r{^((ht|f)tps?://)?}, "#{domain.chomp("/")}/")
     end
 
-    out, _, status= curl_output("--location", "--silent", "--head", "--request", "GET", url.to_s)
-
-    lines = status.success? ? out.lines.map(&:chomp) : []
-
-    locations = lines.map { |line| line[/^Location:\s*(.*)$/i, 1] }
-                     .compact
-
-    redirect_url = locations.reduce(url) do |current_url, location|
```

Comment:
Where has this code moved to?

Reply:
I renamed some variables here to make them clearer (`out` to `headers`, `redirect_url` to `final_url`), so the `Curl#response_status_code_and_location` call that sets `final_url` is replacing this logic.

The intention of `response_status_code_and_location` is to consolidate this logic into one method, as there were separate versions in `Curl#curl_http_content_headers_and_checksum`, `CurlDownloadStrategy#resolve_url_basename_time_file_size`, and `Livecheck::Strategy#page_content`.

The notable difference is that `response_status_code_and_location` uses `URI.join(url, location).to_s` to create absolute URLs instead of the custom logic here. In my testing, `URI#join` handled all of the cases here in the same manner.